lms/models/application_instance.py

```diff
@@ -16,47 +18,137 @@
 
 
 class ApplicationSettings(JSONSettings):
-    fields = (
-        JSONSetting("blackboard", "files_enabled", asbool),
-        JSONSetting("blackboard", "groups_enabled", asbool),
-        JSONSetting("canvas", "sections_enabled", asbool),
-        JSONSetting("canvas", "groups_enabled", asbool),
-        JSONSetting("canvas", "files_enabled", asbool),
-        JSONSetting("canvas", "folders_enabled", asbool),
-        JSONSetting("canvas", "strict_section_membership", asbool),
-        JSONSetting("canvas", "pages_enabled", asbool),
-        JSONSetting("canvas_studio", "admin_email"),
-        JSONSetting("canvas_studio", "client_id"),
-        JSONSetting("canvas_studio", "client_secret", JSONSetting.AES_SECRET),
-        JSONSetting("canvas_studio", "domain"),
-        JSONSetting("desire2learn", "client_id"),
-        JSONSetting("desire2learn", "client_secret", JSONSetting.AES_SECRET),
-        JSONSetting("desire2learn", "groups_enabled", asbool),
-        JSONSetting("desire2learn", "files_enabled", asbool),
-        JSONSetting("google_drive", "files_enabled", asbool),
-        JSONSetting("microsoft_onedrive", "files_enabled", asbool),
-        JSONSetting("moodle", "api_token", JSONSetting.AES_SECRET),
-        JSONSetting("moodle", "groups_enabled", asbool),
-        JSONSetting("moodle", "files_enabled", asbool),
-        JSONSetting("moodle", "pages_enabled", asbool),
-        JSONSetting("vitalsource", "enabled", asbool),
-        JSONSetting("vitalsource", "user_lti_param"),
-        JSONSetting("vitalsource", "user_lti_pattern"),
-        JSONSetting("vitalsource", "api_key"),
-        JSONSetting("vitalsource", "student_pay_enabled", asbool),
-        JSONSetting("jstor", "enabled", asbool),
-        JSONSetting("jstor", "site_code"),
-        JSONSetting("youtube", "enabled", asbool),
-        JSONSetting("hypothesis", "notes", name="Notes"),
-        JSONSetting(
-            "hypothesis",
-            "auto_assigned_to_org",
-            asbool,
-            name="Auto Assigned To Organisation",
-        ),
-        JSONSetting("hypothesis", "instructor_email_digests_enabled", asbool),
-        JSONSetting("hypothesis", "lti_13_sourcedid_for_grading", asbool),
-    )
+    class Settings(StrEnum, Enum):
```

Comment:
This enum + dict of settings it's a bit awkward right now but it helps with the transition.